codegen/src/main.rs

```diff
@@ -155,19 +156,29 @@ impl From<SystemMetadata> for RenderMetadata {
         let mut smart_speaker_service = m.hap.services.get_mut("smart-speaker").unwrap();
         smart_speaker_service.name = "Smart Speaker".to_string();
 
-        let mut sorted_categories = m.homekit.categories.iter().map(|(_, v)| v.clone()).collect::<Vec<_>>();
-        sorted_categories.sort_by(|a, b| a.number.partial_cmp(&b.number).unwrap());
-
-        let mut sorted_characteristics = m.hap.characteristics.iter().map(|(_, v)| v.clone()).collect::<Vec<_>>();
-        sorted_characteristics.sort_by(|a, b| a.name.cmp(&b.name));
-
-        let mut sorted_services = m.hap.services.iter().map(|(_, v)| v.clone()).collect::<Vec<_>>();
-        sorted_services.sort_by(|a, b| a.name.cmp(&b.name));
+        let mut sorted_categories = m
+            .homekit
+            .categories
+            .par_iter()
+            .map(|(_, v)| v.clone())
+            .collect::<Vec<_>>();
+        sorted_categories.par_sort_by(|a, b| a.number.partial_cmp(&b.number).unwrap());
+
+        let mut sorted_characteristics = m
+            .hap
+            .characteristics
+            .par_iter()
+            .map(|(_, v)| v.clone())
+            .collect::<Vec<_>>();
+        sorted_characteristics.par_sort_by(|a, b| a.name.cmp(&b.name));
+
+        let mut sorted_services = m.hap.services.par_iter().map(|(_, v)| v.clone()).collect::<Vec<_>>();
+        sorted_services.par_sort_by(|a, b| a.name.cmp(&b.name));
```

Comment:
Cool! Just out of curiosity, did you benchmark how much faster this got before vs. after rayon? 🤓 Never felt the need for optimizations like this, since you don't codegen that often anyway, but cool!